Parsers/ASimProcessEvent/Parsers/vimProcessTerminateMicrosoftSysmonWindowsEvent.yaml

```diff
@@ -0,0 +1,157 @@
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: | 
+  let parser = (
+      starttime:datetime=datetime(null),
+      endtime:datetime=datetime(null),
+      commandline_has_any:dynamic=dynamic([]),
+      commandline_has_all:dynamic=dynamic([]),
+      commandline_has_any_ip_prefix:dynamic=dynamic([]),
+      actingprocess_has_any:dynamic=dynamic([]),
+      targetprocess_has_any:dynamic=dynamic([]),
+      parentprocess_has_any:dynamic=dynamic([]),
+      actorusername_has:string='*',
+      dvcipaddr_has_any_prefix:dynamic=dynamic([]),
+      dvchostname_has_any:dynamic=dynamic([]),
+      eventtype:string='*',
+      disabled:bool=false
+    ) {
+        let parser_WindowsEvent=(
+            starttime:datetime=datetime(null),
+            endtime:datetime=datetime(null),
+            commandline_has_any:dynamic=dynamic([]),
+            commandline_has_all:dynamic=dynamic([]),
+            commandline_has_any_ip_prefix:dynamic=dynamic([]),
+            actingprocess_has_any:dynamic=dynamic([]),
+            targetprocess_has_any:dynamic=dynamic([]),
+            parentprocess_has_any:dynamic=dynamic([]),
+            actorusername_has:string='*',
+            dvcipaddr_has_any_prefix:dynamic=dynamic([]),
+            dvchostname_has_any:dynamic=dynamic([]),
+            eventtype:string='*',
+            disabled:bool=false
+        ){
+            WindowsEvent
+            | where // pre-filtering
+                (isnull(starttime)   or TimeGenerated >= starttime )
+                and (isnull(endtime) or TimeGenerated <= endtime )
+                and not(disabled)
+                and Provider == "Microsoft-Windows-Sysmon" and EventID == 5
+                and (eventtype=='*' or eventtype=='ProcessTerminated')
+                and (array_length(commandline_has_all)==0 ) 
+                and (array_length(commandline_has_any)==0 ) 
+                and (array_length(commandline_has_any_ip_prefix)==0 ) 
+                and (array_length(actingprocess_has_any)==0 ) 
+                and (array_length(parentprocess_has_any)==0 ) 
+                and (array_length(targetprocess_has_any)==0 or EventData has_any (targetprocess_has_any)) 
+                and (actorusername_has=='*' or EventData has actorusername_has)  
+                and (array_length(dvcipaddr_has_any_prefix)==0)
+                and (array_length(dvchostname_has_any)==0 or Computer has_any (dvchostname_has_any))
+            | extend
+                EventProduct = "Security Events",
+                ActorUsername = tostring(EventData.User),
+                TargetProcessName = tostring(EventData.Image),
+                TargetProcessId = tostring(EventData.ProcessId),
+                TargetProcessGuid = extract ('^{(.*)}$', 1, tostring(EventData.ProcessGuid), typeof(string)),
+                TargetUsername = tostring(EventData.User)
+            | extend TargetUsernameType = iff(isnotempty(TargetUsername),'Windows', ''),
```

Comment:
![image](https://github.com/user-attachments/assets/78602829-f1b6-4d4f-bfe4-1ecda3a4740f)

Reply:
you mean targetusernametype?

Reply:
rolled back changes